src/geckomat/gather_kcats/getStandardKcat.m

```diff
@@ -76,37 +76,52 @@
 % Get the kcat value based on all the kcats in the model
 standardKcat = median(model.ec.kcat(rxnsKcatZero), 'omitnan');
 
-enzSubSystems = cell(numel(model.ec.rxns), 1);
-
-% Get the subSystem for the rxns with a GPR
-for i = 1:numel(model.ec.rxns)
-    if ~model.ec.geckoLight
-        idx = strcmpi(model.rxns, model.ec.rxns{i});
-    else
-        % Remove prefix
-        rxnId = model.ec.rxns{i}(5:end);
-        idx = strcmpi(model.rxns, rxnId); 
+% If the model have subSystems assigned calculate kcat based on subSystem
+if ~isempty([model.subSystems{~cellfun(@isempty, model.subSystems)}])
```

Comment:
Model might not have subSystems field at all.
```suggestion
if isfield(model,'subSystems') & ~isempty([model.subSystems{~cellfun(@isempty, model.subSystems)}])
```

Reply:
I was wondering about that. So, if subSystem is not in conventional GEM, makeEcModel doesn't add it ?